dotCMS/src/main/java/com/dotmarketing/factories/TreeFactory.java

```diff
@@ -1,386 +1,363 @@
+
+            return new DBTreeTransformer(dc.loadObjectResults()).trees();
+        } catch (Exception e) {
+            Logger.warn(TreeFactory.class, "getTree failed:" + e, e);
+        }
+
+        return new ArrayList<Tree>();
+    }
+
+
+
+    private static List<Tree> getTreesByChildAndRelationType(String child, String relationType) {
+        try {
+            DotConnect dc = new DotConnect();
+            dc.setSQL("select * from tree where  child = ? and relation_type = ? order by tree_order asc");
+            dc.addParam(child);
+            dc.addParam(relationType);
+
+            return new DBTreeTransformer(dc.loadObjectResults()).trees();
+        } catch (Exception e) {
+            Logger.warn(TreeFactory.class, "getTree failed:" + e, e);
+        }
+
+        return new ArrayList<Tree>();
+    }
+
+
+    public static List<Tree> getTreesByChildAndRelationType(Inode child, String relationType) {
+        return getTreesByChildAndRelationType(child.getInode(), relationType);
+    }
+
+    public static List<Tree> getTreesByChildAndRelationType(Identifier child, String relationType) {
+        return getTreesByChildAndRelationType(child.getId(), relationType);
+    }
+
+
+    public static List<Tree> getTreesByRelationType(String relationType) {
+        try {
+            DotConnect dc = new DotConnect();
+            dc.setSQL("select * from tree where  relation_type = ?");
+            dc.addParam(relationType);
+
+            return new DBTreeTransformer(dc.loadObjectResults()).trees();
+        } catch (Exception e) {
+            Logger.warn(TreeFactory.class, "getTree failed:" + e, e);
+        }
+
+        return new ArrayList<Tree>();
+    }
+
+
+    public static List<Tree> getTreesByParent(Inode inode) {
+        return getTreesByParent(inode.getInode());
+    }
+
+
+    public static List<Tree> getTreesByParent(String inode) {
+        try {
+            DotConnect dc = new DotConnect();
+            dc.setSQL("select * from tree where  parent = ?");
+            dc.addParam(inode);
+
+            return new DBTreeTransformer(dc.loadObjectResults()).trees();
+        } catch (Exception e) {
+            Logger.warn(TreeFactory.class, "getTree failed:" + e, e);
+        }
+
+        return new ArrayList<Tree>();
+    }
+
+    public static List<Tree> getTreesByChild(Inode inode) {
+        return getTreesByChild(inode.getInode());
+    }
+
+
+    public static List<Tree> getTreesByChild(String inode) {
+        try {
+            DotConnect dc = new DotConnect();
+            dc.setSQL("select * from tree where  child = ?");
+            dc.addParam(inode);
+
+            return new DBTreeTransformer(dc.loadObjectResults()).trees();
+        } catch (Exception e) {
+            Logger.warn(TreeFactory.class, "getTree failed:" + e, e);
+        }
+
+        return new ArrayList<Tree>();
+    }
+
+    public static void swapTrees(final Inode i1, final Inode i2) throws DotDataException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'com.dotmarketing.exception.DotDataException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)